cmd/amp/cli/cli_test.go

```diff
@@ -119,13 +120,15 @@ func runTestSpec(t *testing.T, test *TestSpec) (err error) {
 		startTime := time.Now().UnixNano() / 1000000
 
 		for i = -1; i < cmdSpec.Retry; i++ {
+			err = nil
+
 			cmdString := generateCmdString(&cmdSpec)
 			tmplOutput, tmplErr := performTemplating(strings.Join(cmdString, " "), cache)
 			if tmplErr != nil {
 				err = fmt.Errorf("Executing templating failed: %s", tmplErr)
```

Comment:
I can make the templating error return the same as the timeout err statement, however, the reason the `err = nil` is there is for the regex err. If there is a miss matched output and the command is allowed to retry, it should retry that number of times.